// Proto: Service Definition for gRPC

syntax = "proto3";
package request_handler;

// The Request Handler service definition.
service RequestHandler {

  // Sends Price of the stock if the status is active.
  rpc Lookup (LookupRequest) returns (LookupResponse) {}

  // Executes a Trade for the client and communicates the success of it.
  rpc Trade (TradeRequest) returns (TradeResponse) {}
}

// The request message containing the stock's name.
message LookupRequest {
  string stock_name = 1;
}

// The response message containing the details of the lookup requested.
message LookupResponse {
  int32 success = 1;
  map<string, float> stock_details = 2;
}

// The request message containing the stock's name, volume and buy or sell.
message TradeRequest {
  string stock_name = 1;
  float trade_volume = 2;
  string type = 3;
}

message TradeResponse {
  int32 success = 1;
}